modules/pipeline/pipengine/actionexecutor/plugins/k8sjob/k8sjob.go

```diff
@@ -128,26 +137,79 @@ func (k *K8sJob) Status(ctx context.Context, action *spec.PipelineTask) (desc ap
 		}
 		selector := strutil.Join(matchlabels, ",", true)
 
-		jobPods, err = k.client.ClientSet.CoreV1().Pods(action.Extra.Namespace).List(ctx, metav1.ListOptions{
+		jobPods, err = k.client.ClientSet.CoreV1().Pods(task.Extra.Namespace).List(ctx, metav1.ListOptions{
 			LabelSelector: selector,
 		})
 		if err != nil {
 			return
 		}
 	}
 
-	lastMsg, err := k.getLastMsg(ctx, action.Extra.Namespace, jobName)
+	lastMsg, err := k.getLastMsg(ctx, task.Extra.Namespace, jobName)
 	if err != nil {
 		return
 	}
 
-	//status := generatePipelineStatus(job, jobPods)
-	desc = generateKubeJobStatus(job, jobPods, lastMsg)
-	return
+	status := generateKubeJobStatus(job, jobPods, lastMsg)
+	if status.Status == "" {
+		return desc, errors.Errorf("get empty status from k8sjob, statusCode: %s, lastMsg: %s", status.Status, status.LastMessage)
+	}
+	return apistructs.PipelineStatusDesc{
+		Status: logic.TransferStatus(string(status.Status)),
+		Desc:   status.LastMessage}, nil
 }
 
-func (k *K8sJob) Create(ctx context.Context, action *spec.PipelineTask) (data interface{}, err error) {
-	job, err := logic.TransferToSchedulerJob(action)
+func (k *K8sJob) Exist(ctx context.Context, task *spec.PipelineTask) (created bool, started bool, err error) {
+	statusDesc, err := k.Status(ctx, task)
+	if err != nil {
+		created = false
+		started = false
+		if strutil.Contains(err.Error(), "failed to inspect job, err: not found") {
+			err = nil
+			return
+		}
+		return
+	}
+	return logic.JudgeExistedByStatus(statusDesc)
+}
+
+func (k *K8sJob) Create(ctx context.Context, task *spec.PipelineTask) (data interface{}, err error) {
+	defer logic.WrapTaskErrorForExecutor(k.Kind().String(), &err, "create job", task)
```

Comment:
you should have a default implement for k8s kind.
and overwrite necessary methods.

Reply:
define the k8s base executor interface
impl the `exist,create,remove,batchdelete` for k8s kind
![image](https://user-images.githubusercontent.com/30427474/158511406-96eccc97-99e3-49a7-be5e-4fa808162ee3.png)